app/models/incomeSourceDetails/BusinessDetailsModel.scala

```diff
@@ -36,6 +37,13 @@ case class BusinessDetailsModel(incomeSourceId: String,
   def isCeased: Boolean = cessation.exists(_.date.nonEmpty)
 
   def isOngoingSoleTraderBusiness(id: String): Boolean = !isCeased && id == incomeSourceId
+
+  def getTradingStartDate: LocalDate = (tradingStartDate, contextualTaxYear) match {
+    case (Some(startDate), None) => startDate
+    case (None, Some(taxYear)) => LocalDate.of(taxYear - 1, 4, 6)
+    case (Some(startDate), Some(_)) => startDate
+    case (None, None) => LocalDate.MIN
+  }
```

Comment:
not 100% sure but thinking `getTradingStartDate` can be simplified?
```
def getTradingStartDate: LocalDate = (tradingStartDate, contextualTaxYear) match {
    case (None, None) => LocalDate.MIN
    case (None, Some(taxYear)) => LocalDate.of(taxYear - 1, 4, 6)
    case _ => startDate
  }
```

Reply:
calculation of tax year end date is redundant as its already part of taxYear type


```
  def toFinancialYearStart: LocalDate = {
    LocalDate.of(this.startYear, April, Sixth)
  }

  def toFinancialYearEnd: LocalDate = {
    LocalDate.of(this.endYear, April, Fifth)
  }
```

Reply:
We aren't using the TaxYear class though so I'm not sure if that's relevant here?

Reply:
I would think that next code would be bit easy to ready:

`TaxYear(... ).toFinancialYearStart `